staging/src/k8s.io/apiserver/pkg/server/resourceconfig/helpers.go

```diff
@@ -0,0 +1,214 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourceconfig
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apimachinery/registered"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	serverstore "k8s.io/apiserver/pkg/server/storage"
+	utilflag "k8s.io/apiserver/pkg/util/flag"
+)
+
+// MergeResourceEncodingConfigs merges the given defaultResourceConfig with specific GroupVersionResource overrides.
+func MergeResourceEncodingConfigs(
```

Comment:
What's the use-case for exposing the storage versions?  I don't think we can expect a user to get it right, since exactly which codecs we have is built into the binary.  We started exposing it long ago, but I think it was a mistake.  OpenShift switched over a while back and life has been much better.

Reply:
It's only moved into a package, no new code.

Reply:
> It's only moved into a package, no new code.

It's moved, but it isn't generic.  We shouldn't bring this code forward into the generic apiserver if we can avoid it.  Choosing a storage version isn't a thing that users can do.

Reply:
> It's only moved into a package, no new code.
> It's moved, but it isn't generic. We shouldn't bring this code forward into the generic apiserver if we can avoid it. Choosing a storage version isn't a thing that users can do.

It looks like this comment still stands.